frontend/src/routes/settings/admin/oidc-clients/[id]/+page.svelte

```diff
@@ -49,15 +49,26 @@
 			updatedClient.logo !== undefined
 				? oidcService.updateClientLogo(client, updatedClient.logo)
 				: Promise.resolve();
+		const darkImagePromise =
+			updatedClient.darkLogo !== undefined
+				? oidcService.updateClientDarkLogo(client, updatedClient.darkLogo)
+				: Promise.resolve();
 
 		client.isPublic = updatedClient.isPublic;
 		setupDetails[m.pkce()] = updatedClient.pkceEnabled ? m.enabled() : m.disabled();
 		setupDetails[m.requires_reauthentication()] = updatedClient.requiresReauthentication
 			? m.enabled()
 			: m.disabled();
 
-		await Promise.all([dataPromise, imagePromise])
+		await Promise.all([dataPromise, imagePromise, darkImagePromise])
 			.then(() => {
+				// Update the hasLogo and hasDarkLogo flags after successful upload
+				if (updatedClient.logo !== undefined) {
+					client.hasLogo = updatedClient.logo !== null || !!updatedClient.logoUrl;
+				}
+				if (updatedClient.darkLogo !== undefined) {
+					client.hasDarkLogo = updatedClient.darkLogo !== null || !!updatedClient.darkLogoUrl;
+				}
```

Comment:
See: https://stackoverflow.com/questions/4725603/variable-undefined-vs-typeof-variable-undefined

```suggestion
				if (typeof updatedClient.logo != "undefined") {
					client.hasLogo = updatedClient.logo !== null || !!updatedClient.logoUrl;
				}
				if (typeof updatedClient.darkLogo != "undefined") {
					client.hasDarkLogo = updatedClient.darkLogo !== null || !!updatedClient.darkLogoUrl;
				}
```

Reply:
See below